src/main/java/net/atos/entng/rss/service/ChannelServiceMongoImpl.java

```diff
@@ -34,36 +46,155 @@
-		mongo.delete(collection,  MongoQueryBuilder.build(builder), validResultHandler(handler));
+		Promise<JsonObject> promise = Promise.promise();
+		QueryBuilder builder = QueryBuilder.start(Field.MONGO_ID).is(idChannel);
+		mongo.findOne(collection,  MongoQueryBuilder.build(builder), null, validResultHandler(result -> {
+			if (result.isLeft()) {
+				log.error("[RSS@ChannelServiceMongo::delete] Can't delete user's channel");
+				promise.fail(result.left().getValue());
+				return;
+			}
+			JsonObject channel = result.right().getValue();
+			// can't delete the global channel in this endpoint
+			if (!Boolean.TRUE.equals(channel.getBoolean(Field.GLOBAL, false))) {
+				mongo.delete(collection,  MongoQueryBuilder.build(builder), validResultHandler(PromiseHelper.handler(promise)));
+			} else {
+				log.error("[RSS@ChannelServiceMongo::delete] Can't delete global channel, use /channels/globals");
+				promise.fail("Global channel cannot be deleted");
+			}
+		}));
+		return promise.future();
 	}
 
+	@Override
+	public Future<JsonObject> update(String userId, String idChannel, List<ChannelFeed> feeds) {
+		Promise<JsonObject> promise = Promise.promise();
+		QueryBuilder builder = QueryBuilder.start(Field.MONGO_ID).is(idChannel);
+		mongo.findOne(collection,  MongoQueryBuilder.build(builder), null, validResultHandler(result -> {
+			if (result.isLeft()) {
+				log.error("[RSS@ChannelServiceMongo::update] Can't find user's channel");
+				promise.fail(result.left().getValue());
+				return;
+			}
+			JsonObject channel = result.right().getValue();
+			if (!Boolean.TRUE.equals(channel.getBoolean(Field.GLOBAL, false))) {
+				// get globals channels
+				this.channelGlobalServiceMongo.list()
+					.compose(globalChannels -> ChannelsHelper.filterPreferences(userId, globalChannels, new ArrayList<>()))
+					.onSuccess(userChannels -> {
+						List<ChannelFeed> globalFeeds = ChannelsHelper.removeGlobalFeeds(feeds, userChannels);
+						List<Channel> globalChannelsList = ChannelsHelper.removeGlobalChannels(feeds, userChannels);
+						// remove global feeds
+						feeds.removeAll(globalFeeds);
+						// remove global channels
+						userChannels.removeAll(globalChannelsList);
+						if (!userChannels.isEmpty()) {
+							// add preferences to hidden global channels
+							PreferenceHelper.addPreferences(userId, userChannels)
+								.onFailure(error -> {
+									log.error("[RSS@ChannelServiceMongo::update] Can't add preferences");
+									promise.fail(error);
+								});
+						}
+						// update channel
+						MongoUpdateBuilder modifier = new MongoUpdateBuilder();
+						JsonObject now = MongoDb.now();
+						modifier.set(Field.FEEDS, IModelHelper.toJsonArray(feeds)).set(Field.MODIFIED, now);
+						mongo.update(collection, MongoQueryBuilder.build(builder), modifier.build(), validActionResultHandler(PromiseHelper.handler(promise)));
+					})
+					.onFailure(error -> {
+						log.error("[RSS@ChannelServiceMongo::update] Can't get channels");
```

Comment:
```suggestion
						log.error("[RSS@ChannelServiceMongoImpl::update] Can't get channels");
```
Un peu partout dans le fichier